### DISCUSSION TITLE
Latest version not working on Safari

### DISCUSSION THREAD

**Discussion by icaliman** [alpinejs/alpine#3403] [7. Bugs (archived)]
This code is not working on Safari with `alpinejs@3.11.x`,
but works with `alpinejs@3.10.5`. 

Tested on iPad running iOS 12.5.5.

```html
<div x-data="{ open: false }">
      <button @click="open = true">Expand</button>

      <span x-show="open">
        Content...
      </span>
    </div>
```

**Comment by ekwoka** [comment]
Works fine on Safari 16.3 on macOS.

Would be helpful if you described more what "does not work" means. Like, what part?

Also any logs or anything.

None of those directives were touched between those versions of Alpine.

So without any more info it's hard to diagnose.

**Reply by icaliman** [reply]
For example with this code: https://codesandbox.io/s/alpinejs-w3c-compliant-forked-nsotmv?file=/index.html

The button is not working and the page looks like:
![telegram-cloud-photo-size-2-5407010603714856261-y](https://user-images.githubusercontent.com/4498933/216941049-b3cd9592-846b-46f9-a8cf-30c7901e1680.jpg)

It seems that alpine is not loading at all.

There are no errors in console.

**Reply by ekwoka** [reply]
Interesting, it does work fine on Safari on macOS. without any errors hopefully someone has an ipad to help troubleshoot.

**Reply by icaliman** [reply]
@ekwoka the problem is [at this line with **??**](https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives.js#L31)

The CDN build contains `??` operator which is not supported in older browsers.

I think at this line there is a bug, because if an item is not found `indexOf` returns `-1`, not `null`. And `-1 ?? SOMETHING_ELSE` will be always `-1`.

**Reply by ekwoka** [reply]
It's a new enough feature (safari on ios has supported it for nearly 3 years (and if you need to target further back, you can put those settings into your bundler on your project)

but yes, it should not be there like that. It's not used in the core, but for plugins. but the browser. Good catch

Made a PR to fix it, but the issue of support for old version of safari is on the developer using Alpine. any bundler can transform the nullish coalescing operator when you target older versions if you need to. This PR will remove this instance of `??` to fix the intent, but the mask plugin uses it, and there is no guarantee it won't be used in future releases.
